exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/metadata/Metadata_V4.java

```diff
@@ -400,11 +405,23 @@ public Object deserializeKey(String key, com.fasterxml.jackson.databind.Deserial
    * Note: Since the structure of column metadata hasn't changes from v3, ColumnMetadata_v4 extends ColumnMetadata_v3
    */
   public static class ColumnMetadata_v4 extends Metadata_V3.ColumnMetadata_v3 {
+
+    List<OriginalType> originalTypes;
+
     public ColumnMetadata_v4() {
     }
 
     public ColumnMetadata_v4(String[] name, PrimitiveType.PrimitiveTypeName primitiveType, Object minValue, Object maxValue, Long nulls) {
+      this(name, primitiveType, minValue, maxValue, nulls, new ArrayList<>());
+    }
+
+    public ColumnMetadata_v4(String[] name, PrimitiveType.PrimitiveTypeName primitiveType, Object minValue, Object maxValue, Long nulls, List<OriginalType> originalTypes) {
       super(name, primitiveType, minValue, maxValue, nulls);
+      this.originalTypes = Collections.unmodifiableList(originalTypes);
```

Comment:
`this.parentTypes = new ArrayList<>(parentTypes);` you wrap into list, here you wrap in unmodifiable... looks like different approach for similar cases...